src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -267,6 +275,13 @@ def upload_stream(
                     )
                     raise ne.with_traceback(tb) from None
             else:
+                if SNOWFLAKE_CONNECTOR_VERSION >= (3, 0, 3, None):
+                    kwargs = {
```

Comment:
Same as above, I prefer explicitly specifying the parameters. Is there a reason why we would want to use a dictionary?